### PR TITLE
Added required fields for Node API Parity

### LINKED ISSUE #3135: Adding support for node.retired and node.retired_reason
This is a feature that I would like to have access too.

Gophercloud code being changed: https://github.com/gophercloud/gophercloud/blob/d8427580463e54c7738e5499ce234417ce8223ff/openstack/baremetal/v1/nodes/results.go#L94

The fields as defined in Ironic: https://github.com/openstack/ironic/blob/1e52143f07266b53ad47d49a5be04cbf85cccc8b/ironic/objects/node.py#L178

### PR DESCRIPTION
**Pull request by Sharpz7**
Fixes https://github.com/gophercloud/gophercloud/issues/3172

I have 1-1 mapped with: https://docs.openstack.org/api-ref/baremetal/#list-nodes-detailed

=========================

Since all the fields I have added were already present in the test fixtures, this was an easy change!

### CODE REVIEW

File: openstack/baremetal/v1/nodes/results.go [hunk 1]
```diff
@@ -183,8 +185,16 @@ type Node struct {
 	// Current deploy step.
 	DeployStep map[string]any `json:"deploy_step"`
 
-	// Current service step.
-	ServiceStep map[string]any `json:"service_step"`
+	Links []Link `json:"links"`
```
Comment (EmilienM) [inline openstack/baremetal/v1/nodes/results.go:188]: Missing doc string

File: openstack/baremetal/v1/nodes/results.go [hunk 2]
```diff
@@ -90,12 +90,14 @@ func (r SubscriptionVendorPassthruResult) Extract() (*SubscriptionVendorPassthru
 	return &s, err
 }
 
+type Link struct {
+	Href string `json:"href"`
```
Comment (EmilienM) [inline openstack/baremetal/v1/nodes/results.go:94]: Missing doc string

File: openstack/baremetal/v1/nodes/results.go [hunk 3]
```diff
@@ -90,12 +90,14 @@ func (r SubscriptionVendorPassthruResult) Extract() (*SubscriptionVendorPassthru
...
+	Rel  string `json:"rel"`
```
Comment (EmilienM) [inline openstack/baremetal/v1/nodes/results.go:95]: Missing doc string

**Comment by EmilienM** [comment]
also please squash

**Comment by EmilienM** [comment]
I tagged it for a v2 backport candidate since I'm sure you'll need that downstream.

**Comment by Sharpz7** [comment]
That should be it. Merci :))

**Comment by EmilienM** [comment]
@Sharpz7 squash it please

**Comment by Sharpz7** [comment]
@EmilienM I think I have? its a single commit?

**Review by EmilienM — APPROVED** [review]

**Pull request merged**